pkg/clusterd/disk.go

```diff
@@ -66,6 +67,24 @@ func DiscoverDevices(executor exec.Executor) ([]*sys.LocalDisk, error) {
 			// go on without udev info
 			logger.Warningf("failed to get udev info for device %s: %+v", d, err)
 		}
+
+		// Test if device has child, if so we skip it and only consider the partitions
+		// which will come in later iterations of the loop
+		// We only test if the type is 'disk', this is a property reported by lsblk
+		// and means it's a parent block device
+		if disk.Type == sys.DiskType {
+			deviceChild, err := sys.ListDevicesChild(executor, d)
+			if err != nil {
+				logger.Warningf("failed to detect child devices for device %q, assuming they are none. %+v", d, err)
+			}
+			// lsblk will output at least 2 lines if they are partitions, one for the parent
+			// and N for the child
+			if len(deviceChild) > 1 {
+				logger.Infof("skipping device %q because it has child, considering the child instead.", d)
+				continue
```

Comment:
From this line, the `disk` will not be appended to `disks`. Should we be appending the child partitions to `disks` here before continuing? Or do child partitions show up in the range of `devices` being looped over?

Reply:
The child will show up in one of the next iterations so we just want to continue.